TYPO3.Flow/Classes/TYPO3/Flow/Persistence/Doctrine/EntityManagerFactory.php

```diff
@@ -97,6 +97,17 @@ public function create()
         $config->setProxyNamespace('TYPO3\Flow\Persistence\Doctrine\Proxies');
         $config->setAutoGenerateProxyClasses(false);
 
+        // Set default host to 127.0.0.1 if there is no other host configured and at least one other necessary option
+        if ($this->settings['backendOptions']['host'] == '') {
+            if (
+                $this->settings['backendOptions']['dbname'] != '' ||
+                $this->settings['backendOptions']['user'] != '' ||
+                $this->settings['backendOptions']['password'] != ''
+            ) {
+                $this->settings['backendOptions']['host'] = '127.0.0.1';
+            }
+        }
+
```

Comment:
We should always use type safe comparison according to the [CGL](http://flowframework.readthedocs.org/en/latest/TheDefinitiveGuide/PartV/CodingGuideLines/PHP.html#php-in-general). And `password` could be left empty intentionally (not a good idea, but...).
Also, why nest the conditions?
Couldn't something like this work out:
```
if (empty($this->settings['backendOptions']['host']) && !empty($this->settings['backendOptions']['dbname'])) {
  $this->settings['backendOptions']['host'] = '127.0.0.1';
}
```
?